### PR TITLE
packages: add ecr-credential-provider-1.32 and beta0 source for kubernetes-1.32

### PR DESCRIPTION
**Pull request by Sparksssj**
**Issue number:**

Part of https://github.com/bottlerocket-os/bottlerocket/issues/4318

**Description of changes:**

Initial set of changes for kubernetes 1.32 and ecr-credential-provider 1.32 support.

**Testing done**

Manual checks that both packages build successfully.

```
$ rpm -qpi build/rpms/kubernetes-1.32/bottlerocket-kubelet-1.32-1.32.0-0.beta0.1734378736.9d11ae11.br1.x86_64.rpm 
Name        : bottlerocket-kubelet-1.32
Version     : 1.32.0
Release     : 0.beta0.1734378736.9d11ae11.br1
Architecture: x86_64
Install Date: (not installed)
Group       : Unspecified
Size        : 1503031
License     : Apache-2.0 AND BSD-3-Clause
Signature   : (none)
Source RPM  : bottlerocket-kubernetes-1.32.0-0.beta0.1734378736.9d11ae11.br1.src.rpm
Build Date  : Mon Dec 16 19:56:41 2024
Build Host  : localhost
URL         : https://github.com/kubernetes/kubernetes
Summary     : Container cluster node agent
Description :
Container cluster node agent.

$ rpm -qpi build/rpms/ecr-credential-provider-1.32/bottlerocket-ecr-credential-provider-1.32-1.32.0-1.1734378736.9d11ae11.br1.x86_64.rpm 
Name        : bottlerocket-ecr-credential-provider-1.32
Version     : 1.32.0
Release     : 1.1734378736.9d11ae11.br1
Architecture: x86_64
Install Date: (not installed)
Group       : Unspecified
Size        : 1005122
License     : Apache-2.0
Signature   : (none)
Source RPM  : bottlerocket-ecr-credential-provider-1.32-1.32.0-1.1734378736.9d11ae11.br1.src.rpm
Build Date  : Mon Dec 16 20:00:44 2024
Build Host  : localhost
URL         : https://github.com/kubernetes/cloud-provider-aws
Summary     : Amazon ECR credential provider
Description :
Amazon ECR credential provider.
```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by koooosh** [comment]
nit: For the `package` commits, we technically are not updating the packages; instead we are adding new ones, so I prefer:
- `packages: add kubernetes-1.32`
- `packages: add ecr-credential-provider-1.32`

File: packages/ecr-credential-provider-1.32/ecr-credential-provider-1.32.spec [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+%global goimport %{goproject}/%{gorepo}
+
+%global gover 1.32.0
+%global rpmver %{gover}
+
+%global _dwz_low_mem_die_limit 0
+
+Name: %{_cross_os}ecr-credential-provider-1.32
+Version: %{rpmver}
+Release: 0.beta1%{dist}
```
Comment (koooosh) [inline packages/ecr-credential-provider-1.32/ecr-credential-provider-1.32.spec]: Can we run `rpm -qpi ` on one of these rpms to verify the versioning looks good? Let's add this to the PR testing description.

**Comment by koooosh** [comment]
Even though the ecr-credential-provider-1.32 is technically not a beta source, I'm okay with referring to it as such since[ the Latest release is still 1.31.4](https://github.com/kubernetes/cloud-provider-aws/releases/tag/v1.31.4).

File: packages/kubernetes-1.32/kubernetes-1.32.spec [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+# multiarch/qemu-user-static ignored, we're not using it
+cp third_party/forked/gonum/graph/LICENSE LICENSE.gonum.graph
+cp third_party/forked/shell2junit/LICENSE LICENSE.shell2junit
+cp third_party/forked/golang/LICENSE LICENSE.golang
+cp third_party/forked/golang/PATENTS PATENTS.golang
+
+%build
+export FORCE_HOST_GO=1
+
+export GO_VERSION="1.23.3"
```
Comment (yeazelm) [inline packages/kubernetes-1.32/kubernetes-1.32.spec]: Can we make this just GO_MAJOR instead of pinning to a specific version?

**Comment by koooosh** [comment]
^ Force push updates `GO_VERSION` var to a more generic `GO_MAJOR `

**Review by yeazelm — APPROVED** [review]

File: packages/kubernetes-1.32/kubernetes-1.32.spec [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+# KUBE_LINUX_AMD64_CC=x86_64-bottlerocket-linux-gnu-gcc
+# KUBE_LINUX_ARM64_CC=aarch64-bottlerocket-linux-gnu-gcc
+%global kube_cc %{shrink: \
+  %{lua: print(string.upper( \
+     rpm.expand("KUBE_%{_cross_go_os}_%{_cross_go_arch}_CC=")) .. \
+     rpm.expand("%{_cross_target}-gcc")) }}
+
+Name: %{_cross_os}%{gorepo}
+Version: %{rpmver}
+Release: 0.beta1%{dist}
```
Comment (bcressey) [inline packages/kubernetes-1.32/kubernetes-1.32.spec]: Your source artifact says it's "beta0", not "beta1".

File: packages/ecr-credential-provider-1.32/ecr-credential-provider-1.32.spec (hunk 1 above)
Comment (bcressey) [inline packages/ecr-credential-provider-1.32/ecr-credential-provider-1.32.spec]: It's weird to label this the "beta1" release when it's the actual upstream 1.32.0 release. If you needed to put out a revised version of this package while keeping the upstream release the same, you would increment the release field instead.

File: packages/ecr-credential-provider-1.32/ecr-credential-provider-1.32.spec [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+Epoch: 1
```
Comment (bcressey) [inline packages/ecr-credential-provider-1.32/ecr-credential-provider-1.32.spec]: not needed:
```suggestion
```

File: packages/kubernetes-1.32/kubernetes-1.32.spec [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+Epoch: 1
```
Comment (bcressey) [inline packages/kubernetes-1.32/kubernetes-1.32.spec]:
```suggestion
```

File: packages/ecr-credential-provider-1.32/ecr-credential-provider-1.32.spec (hunk 1 above)
Comment (koooosh) [reply packages/ecr-credential-provider-1.32/ecr-credential-provider-1.32.spec]: Discussed offline: we should not enforce arbitrary versioning and we prefer to match the source. Will change this to the official release: `1%{dist}`

File: packages/ecr-credential-provider-1.32/ecr-credential-provider-1.32.spec (hunk 4 above)
Comment (koooosh) [reply packages/ecr-credential-provider-1.32/ecr-credential-provider-1.32.spec]: Discussed offline: Before, the convention was to add `Epoch` for package versions less than the core-kit version, but that only is necessary for existing packages, so we can remove this here.

**Comment by koooosh** [comment]
^ Force push updates versioning fields in specs. It removes the `Epoch` field and updates the `Release` field to match source.

**Review by bcressey — APPROVED** [review]

**Pull request merged**